### PR TITLE
Replace GcsPath by IOChannelFactory in WordCount.

### PR DESCRIPTION
**Pull request by peihe**

### CODE REVIEW

**Comment by peihe** [comment]
R: @dhalperi

File: examples/java/src/main/java/org/apache/beam/examples/WordCount.java [hunk 1]
```diff
@@ -176,9 +179,10 @@ public String apply(KV<String, Long> input) {
     public static class OutputFactory implements DefaultValueFactory<String> {
       @Override
       public String create(PipelineOptions options) {
-        if (options.getTempLocation() != null) {
-          return GcsPath.fromUri(options.getTempLocation())
-              .resolve("counts.txt").toString();
+        String tempLocation = options.getTempLocation();
+        if (!Strings.isNullOrEmpty(tempLocation)) {
+          OChannelFactory factory = IOChannelUtils.getFactory(tempLocation);
```
Comment (dhalperi) [inline examples/java/src/main/java/org/apache/beam/examples/WordCount.java]: typo: IOChannelFactory

Comment (peihe) [inline examples/java/src/main/java/org/apache/beam/examples/WordCount.java]: fixed

**Comment by dhalperi** [comment]
Travis still failing -- looks like there's a checked exception issue. Please take another look.

**Comment by peihe** [comment]
fixed the build

**Comment by dhalperi** [comment]
LGTM

**Pull request closed without merging**